Interpreter_PrimArith.c

```diff
@@ -0,0 +1,236 @@
+#include "Interpreter.h"
+#include "Interpreter_PrimArith.h"
+#include <stdio.h>
+
+Bool Interpreter_dispatchArithmeticPrimitives( int primitiveIndex ) {
+	if( primitiveIndex < 20 ) {
+		return Interpreter_dispatchIntegerPrimitives( primitiveIndex );
+	}
+/*	if( primitiveIndex < 40 ) {
+		return Interpreter_dispatchLargeIntegerPrimitives( primitiveIndex );
+	}
+	if( primitiveIndex < 60 ) {
+		return Interpreter_dispatchFloatPrimitives( primitiveIndex );
+	}*/
+	return NO;
+}
+
+Bool Interpreter_dispatchIntegerPrimitives( int primitiveIndex ) {
+	switch( primitiveIndex ) {
+		case 1: return Interpreter_primitiveAdd();
+		case 2: return Interpreter_primitiveSubtract();
+		case 3: return Interpreter_primitiveLessThan();
+		case 4: return Interpreter_primitiveGreaterThan();
+		case 5: return Interpreter_primitiveLessOrEqual();
+		case 6: return Interpreter_primitiveGreaterOrEqual();
+		case 7: return Interpreter_primitiveEqual();
+		case 8: return Interpreter_primitiveNotEqual();
+		case 9: return Interpreter_primitiveMultiply();
+		case 10: return Interpreter_primitiveDivide();
+		case 11: return Interpreter_primitiveMod();
+		case 12: return Interpreter_primitiveDiv();
+		case 13: return Interpreter_primitiveQuo();
+		case 14: return Interpreter_primitiveBitAnd();
+		case 15: return Interpreter_primitiveBitOr();
+		case 16: return Interpreter_primitiveBitXor();
+		case 17: return Interpreter_primitiveBitShift();	
+		case 18: return Interpreter_primitiveMakePoint();	
+	}
+	return NO;
+}
+
+Bool Interpreter_primitiveAdd(void) {
+	short integerArgument = Interpreter_popInteger();
+	short integerReceiver = Interpreter_popInteger();
+	short integerResult = 0;
+	if( Interpreter_success() == YES ) {
+		integerResult = integerReceiver + integerArgument;
+		Interpreter_success_( ObjectMemory_isIntegerValue(integerResult ) );
```

Comment:
Interpreter_PrimArith.c: In function ‘Interpreter_primitiveAdd’:
Interpreter_PrimArith.c:48:25: warning: implicit declaration of function ‘ObjectMemory_isIntegerValue’ [-Wimplicit-function-declaration]
   48 |   Interpreter_success_( ObjectMemory_isIntegerValue(integerResult ) );
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~